cmd/query/app/querysvc/query_service.go

```diff
@@ -40,15 +42,15 @@ type StorageCapabilities struct {
 
 // QueryService contains span utils required by the query-service.
 type QueryService struct {
-	spanReader       spanstore.Reader
+	traceReader      tracestore.Reader
```

Comment:
what about archive reader/writer, any reason not to upgrade them too? (or PR is too large?)

Reply:
@yurishkuro I initially thought that we couldn't do this yet because I thought we were using the `ArchiveFactory` interface here. However, it seems like we just use the normal reader interface on the archive reader as well (see https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/querysvc/query_service.go#L69) so this can be done. I can do it in this PR or a separate one. Let me know if you have a preference.

Reply:
Because of the other question (not having v2 reader fully implemented) it's ok to do these separately, no particular benefit afaict.